src/services/menu_creator_manager.py

```diff
@@ -1386,10 +1399,21 @@ def create_save_menu(save_game_function: Callable) -> InfoBox:
     element_grid = []
 
     for i in range(SAVE_SLOTS):
+    # Look through the save files for presence of the timestamp tag, and if found include below the save slot title.
```

Comment:
The duplicated portion of code starts to get bigger, let's extract it to a private method such as `_generate_saves_grid` taking a lambda as argument for the callback (since the button callback is different for save/load).